lib/logstash/plugin_mixins/event_support/event_factory.rb

```diff
@@ -0,0 +1,42 @@
+require 'logstash/event'
+
+module LogStash; module PluginMixins; module EventSupport
+
+  class EventFactory
```

Comment:
consider a new design for `EventFactory` where they get the target name-space.
we will need that use-case for the json codecs (check whether there's other cases)